src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf16.cs

```diff
@@ -543,12 +543,37 @@ public void WriteValue(Guid value)
         /// </summary>
         public void WriteNull()
         {
-            //TODO: Optimize, just like WriteValue(long value)
-            WriteItemSeperatorUtf16();
-            _firstItem = false;
-            WriteSpacingUtf16();
+            int charsNeeded = (_firstItem ? 0 : 1) + (_prettyPrint ? 2 + (_indent + 1) * 2 : 0);
+            int bytesNeeded = charsNeeded * 2 + JsonConstants.NullValueUtf16.Length;
+            Span<byte> byteBuffer = EnsureBuffer(bytesNeeded);
+            Span<char> charBuffer = MemoryMarshal.Cast<byte, char>(byteBuffer);
```

Comment:
Is EnsureBuffer _guaranteed_ to return a span that is 16-bit aligned? This cast has undefined behavior if the input span isn't properly aligned for the destination type.

Reply:
EnsureBuffer does not ensure the span is 16-bit aligned, but in this particular use case, it should, because we are manipulating a character stream here. Is there a way I can write an assert to ensure it is aligned?